extensions/wikia/UserProfilePageV3/maintenance/migrateAvatarsViaService.php

```diff
@@ -0,0 +1,224 @@
+	}
+
+	public function execute() {
+		$this->output( date('r') . "\n" );
+
+		// read options
+		$this->isDryRun = $this->hasOption( 'dry-run' ) || !$this->hasOption( 'force' );
+
+		if ( $this->isDryRun ) $this->output( "Running in dry-run mode!\n\n" );
+
+		$this->output( "Getting the list of all accounts...\n" );
+
+		// get all accounts
+		$db = $this->getDB( DB_SLAVE );
+
+		$res = $db->select(
+			'`user`',
+			'user_id AS id',
+			[],
+			__METHOD__
+		);
+
+		$rows = $res->numRows();
+		$this->output( "Processing {$rows} users...\n" );
+
+		$this->output( "Will start in 5 seconds...\n" );
+		sleep( 5 );
+
+		// process users
+		foreach ( $res as $n => $row ) {
+			try {
+				$user = User::newFromId( $row->id );
+				$this->output( sprintf( "\n%d (%.2f%%) [User #%d / %s]: ", ( $n + 1 ), ( ( $n + 1 ) / $rows * 100 ), $user->getId(), $user->getName() ) );
+				$this->processUser( $user );
+			}
+			catch ( AvatarsMigratorException $e ) {
+				\Wikia\Logger\WikiaLogger::instance()->error( __CLASS__, [
+					'exception' => $e,
+					'user_id'   => $user->getId(),
+				] );
+
+				$this->output( sprintf( "\n\t%s", $e->getMessage() ) );
+			}
+		}
+
+		$this->output( "Dome!\n" );
+		$this->output( date('r') . "\n" );
+	}
+
+	/**
+	 * Migrate the avatar of a given user
+	 *
+	 * @param User $user
+	 * @throws AvatarsMigratorException
+	 */
+	private function processUser( User $user ) {
+		$avatar = $user->getGlobalAttribute( AVATAR_USER_OPTION_NAME );
+
+		// no avatar set, skip this account
+		if ( is_null( $avatar ) ) {
+			$this->output( 'no avatar set - skipping' );
+			return;
+		}
+
+		// no need to store default avatars in user properties - remove the entry
+		else if ( self::isDefaultAvatar( $avatar ) ) {
+			$this->output( 'default avatar set - removing an attribute' );
+
+			$this->setAvatarUrl( $user, '' );
+		}
+		// predefined avatar (Avatar*.jpg)
+		else if ( self::isPredefinedAvatar( $avatar ) ) {
+			// store the full URL in user properties
+			$masthead = Masthead::newFromUser( $user );
+			$avatarUrl = $masthead->getPurgeUrl(); # e.g. http://images.wikia.com/messaging/images//1/19/Avatar.jpg
+
+			$this->output( "predefined avatar set <{$avatar}> - setting a full URL: <{$avatarUrl}>" );
+
+			$this->setAvatarUrl( $user, $avatarUrl );
+		}
+		// custom, old avatar - upload via avatars service
+		else if ( !self::isNewAvatar( $avatar ) ) {
+			$service = new UserAvatarsService( $user->getId() );
+
+			// fetch the user-uploaded avatar
+			$masthead = Masthead::newFromUser( $user );
+			$avatarUrl = $masthead->getPurgeUrl();
+
+			$this->output( sprintf( 'uploading <%s>', $avatarUrl ) );
+
+			if ( $this->isDryRun ) return;
+
+			$avatarContent = Http::get( $avatarUrl, 'default', [ 'noProxy' => true ] );
+			if ( empty( $avatarContent ) ) {
+				throw new AvatarsMigratorException( 'Avatar fetch failed' );
+			}
+
+			// store it in temporary file
+			$tmpFile = tempnam( wfTempDir(), 'avatar' );
```

Comment:
Is it worth doing some randomization here to prevent race conditions (two migrations running at the same time or something)?